src/shared/logger/activation.ts

```diff
@@ -198,3 +202,63 @@ async function registerLoggerCommands(context: vscode.ExtensionContext): Promise
         )
     )
 }
+
+/**
+ * Deletes the older logs when there are too many or are using too much space.
+ */
+async function cleanLogFiles(): Promise<void> {
+    const logsDir = path.join(ext.context.globalStoragePath, 'logs')
+    let files = await fs.readdir(logsDir)
+
+    if (files.length > 100) {
+        sortThenDeleteLogs(logsDir, files, 10)
+        files = await fs.readdir(logsDir)
+    }
+
+    let dirSize = 0
+    const oversizedFiles = []
+    for (const log of files) {
+        const logSize = (await fs.stat(path.join(logsDir, log))).size
+        dirSize += logSize
+        if (logSize > ONE_HUNDRED_MEGABYTES) {
+            oversizedFiles.push(log)
+        }
+    }
+    // remove any single files over 100MB
+    if (oversizedFiles.length) {
+        for (const file of oversizedFiles) {
+            await fs.unlink(path.join(logsDir, file))
+        }
+        files = await fs.readdir(logsDir)
+    }
+    if (dirSize > ONE_HUNDRED_MEGABYTES) {
+        sortThenDeleteLogs(logsDir, files, 2)
+    }
+}
+
+/**
+ * Deletes the oldest created files, leaving the desired quantity of latest files.
+ */
+async function sortThenDeleteLogs(logsDir: string, files: string[], keepLatest: number): Promise<void> {
+    files.sort((a, b) => {
+        return fs.statSync(path.join(logsDir, a)).birthtimeMs - fs.statSync(path.join(logsDir, b)).birthtimeMs
```

Comment:
- fs.statSync() can fail.
- calling fs.statSync() potentially 2*100 times on startup is a cost we might as well avoid if we can.
- we already know that the log files are named in big-endian datetime format. So we don't need to check the birthtime. Just sort by filename.
- birthtime is not supported on all OSes: https://nodejs.org/api/fs.html#fs_stat_time_values

Reply:
We may still need to call fs.stat to verify we have a file and not a directory before we try and remove it. (see latest commit)

My original idea was to sort by filename, but thought birthtime would be safer in case we changed the naming of our log files.  Not being supported by other OSes has changed my mind.

Reply:
> We may still need to call fs.stat to verify we have a file and not a directory before we try and remove it.

fs.unlink() won't remove a directory, so we can just rely on that. Avoids some noise in our code.